### ISSUE TITLE
Exit with non-zero code on Cannot parse selector syntax error

### ISSUE THREAD

**Issue by FezVrasta** [stylelint/stylelint#2545]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When stylelint raises a "Cannot parse selector" warning, I would like it to return with a non zero code to make the linting fail.

> What CSS is needed to reproduce this issue?

```css
data-something="true"] .test
  color: red
```

> What stylelint configuration is needed to reproduce this issue?

No configuration needed.

> Which version of stylelint are you using?

`7.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint "**/*.sss"`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

I don't think so.

> What did you expect to happen?

An error should be thrown.

> What actually happened (e.g. what warnings or errors you are getting)?

No error, just a log in the output.

**Comment by jeddy3** [comment]
@FezVrasta Thank you for the report and for using the template.

I can reproduce this with standard CSS:

```css
data-something="true"] .test {
  color: red;
}
```

Gives:

```shell
jeddy3:stylelint-test jeddy3$ ./node_modules/.bin/stylelint sources/css/exit.css

sources/css/exit.css
 1:1     Cannot parse selector

jeddy3:stylelint-test jeddy3$ echo $?
0
```

Whereas a PostCSS syntax error (rather than the `postcss-selector-parser` error above):

```css
data-something="true"] .test {
  color: red;
```

Gives:

```
jeddy3:stylelint-test jeddy3$ ./node_modules/.bin/stylelint sources/css/exit.css

sources/css/exit.css
 1:1  ✖  Unclosed block   CssSyntaxError

jeddy3:stylelint-test jeddy3$ echo $?
2
```

Should the behaviour for both syntax errors be consistent i.e. throw an error?

**Comment by FezVrasta** [comment]
Yes it would be Ideal. On SugarSS PostCSS doesn't throw any error

**Comment by jeddy3** [comment]
@stylelint/core Anyone got any ideas about how this could be implemented?

I believe the `Cannot parse selector` warning in produced in [`parseSelector.js`](https://github.com/stylelint/stylelint/blob/master/lib/utils/parseSelector.js) and [`selector-max-specificity`](https://github.com/stylelint/stylelint/blob/master/lib/rules/selector-max-specificity/index.js#L67).

**Comment by jeddy3** [comment]
Unless this issue is a personal itch that one of the core team wishes to implement, I'm happy bumping this to the next `future-major` milestone. I think it's OK to have feature requests labelled as "help wanted" in the issue tracker until it becomes an personal itch a contributors wants to fix themselves.

Going forward, I also think we can do a major release whenever a bug fix or feature demands it, rather than grouping issues into one prelonged release. So, bumping issues out of milestones becomes less of a big thing.

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/issues/2713
